_topics/en/pooled-mining.md

```diff
@@ -0,0 +1,216 @@
+
+## Pool protocols
+
+The earliest pool protocol was built around transmitting block header
+templates in the format used by the early Bitcoin Core RPC `getwork`.
+When the speed of mining equipment made that impractical, an early
+attempt at creating a distributed mining protocol was specified in BIPs
+[22][BIP22] and [23][BIP23], with a partial implementation in Bitcoin
+Core as the `getblocktemplate` RPC.  GetBlockTemplate never saw
+widespread usage as a pool protocol itself, although many centralized
+pool servers have used it for the past decade in the background to obtain
+a good set of transactions to mine from their local Bitcoin Core node.
+
+As the GetWork-based protocol became problematic, most mining pools
+switched to the Stratum protocol (v1).  Stratum v1 allows pools to
+only send their members a template for a block header and a coinbase
+transaction, which can be less than 200 bytes.  When a member finds a
+share, they can send back roughly the same information.  This very
+compact information minimizes the overhead of the pool protocol.  The
+downside of the way GetBlock and Stratum v1 are typically used is that
+pool members have no insight or direct control what they mine.  They
+aren't directly informed about which transactions are included in the
+template block and the pool may reject any attempt they make to change
+those transactions.  They are (by necessity) informed of what their
+block claims is the previous block header, but few Stratum v1 users
+ensure that is what their own full nodes think should be the previous
+block header; this lack of independent validation has allowed pools in
+the past to [steal][betcoindice] from website operators and lose money
+accidentally [violating consensus rules][jul2015 chainsplits].  Although
+BIPs [40][bip40] and [41][bip41] were reserved for Stratum v1,
+documentation for them has never been provided.
+
+[Betterhash][] was a proposed mining pool protocol that allowed pool
+members to select which transactions to include in the blocks they
+create.  The protocol also made it easy for pool members who found a
+new block to submit it directly to the network through their Bitcoin
+full node, which could reduce propagation times and improve the chance
+of that block becoming a permanent part of the block chain.  Additional
+improvements in Betterhash focused on security and efficiently
+distributing work across multiple mining hardware devices.  Although the
+proposal has not been withdrawn, it does appear that development of it
+has been replaced by development on Stratum v2.
+
+[Stratum v2][] is an entirely new protocol developed by several of the
+same people who contributed to Stratum v1.  It provides several of the
+same advantages as Betterhash, although sometimes using different
+mechanisms.  Like Betterhash, one of its key advantages is that it can
+allow individual pool members to chose which transactions to include in
```

Comment:
```suggestion
mechanisms.  Like Betterhash, one of its key advantages is that it
allows individual pool members to choose which transactions to include in
```
(unless Stratum v2 can be run in a way that _doesn't_ allow members to choose transactions?)

Reply:
> ```suggestion
> mechanisms.  Like Betterhash, one of its key advantages is that it
> allows individual pool members to choose which transactions to include in
> ```
> (unless Stratum v2 can be run in a way that _doesn't_ allow members to choose transactions?)

SV2 is a modular protocol, so it could be configured and used with encryption and binary framing capabilities, but without allowing individual miners to create and work on custom block templates. Of course the best configuration is the one in which miners select txs and work on custom block templates.
To dig more into it, every possible protocol scenario can be found at stratumprotocol.org

Reply:
Leaving text as-is.  I wrote this in the hypothetical voice becaues, as @GitGab19 says, pools can disallow independent transaction selection (either through the protocol or though simply not paying for any shares submitted with unapproved transactions).